src/string.cr

```diff
@@ -1743,6 +1743,21 @@ class String
     end
   end
 
+  # Relturns a string where all strings in the given hash are replaced.
+  # by the corresponding *hash* values.
+  #
+  # ```
+  # "hello".gsub({"e" => 'a', "l" => 'd'}) # "haddo"
+  # ```
+  def gsub(hash : Hash(String, _))
+    string = self
+    hash.each do |key, value|
+      string = string.gsub(key, value)
```

Comment:
I think that iterating the string for every replacement is not the best performing solution. You probably want to find the last character of every key, and place them in an array, then iterate the string checking against each character. If a match is found perform strcmp calls to determine if any of the full strings match.